src/Messaging/MessageDataAssertion.php

```diff
@@ -82,8 +82,8 @@ private static function assertSubPayload($payload)
         if (is_array($payload)) {
             foreach ($payload as $subPayload) {
                 self::assertSubPayload($subPayload);
-                return;
             }
+            return;
```

Comment:
damn, the coverage report was right. the implementation was wrong. Didn't notice it before. Good catch :+1: